### DISCUSSION TITLE
Foundry Fund Me: Tests Fail with Forked zkSync Sepolia Node

### DISCUSSION THREAD

**Discussion by spencerj411** [Cyfrin/foundry-full-course-cu#2178] [Q&A] [answered]
Hi, Cyfrin Updraft fam! Maybe I'm getting way ahead of myself. For fun, I added zkSync Sepolia in my HelperConfig; I got the chain id of zksync sepolia as well as the price feed address of the reference contract on zksync sepolia. The below is the code snippet:
```
else if (block.chainid == 300) {
            console.log("ZKSYNC SEPOLIA L2 ROLLUP BLOCKCHAIN DETECTED");
            // zkSync Sepolia testnet (L2 rollup)
            activeConfig = NetworkConfig({
                priceFeedAddress: 0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF
            });
        }
```
I tried running the tests using a forked zkSync sepolia node (via alchemy):
`forge test --fork-url "https://zksync-sepolia.g.alchemy.com/v2/....." -vvvv` (both on vanilla foundry and foundry zksync)
5/7 tests failed:
```
[FAIL. Reason: EvmError: Revert] testCorrectFund() (gas: 19934)
[FAIL. Reason: EvmError: Revert] testFundersList() (gas: 19979)
[FAIL. Reason: EvmError: Revert] testGetVersion() (gas: 7987)
[FAIL. Reason: EvmError: Revert] testOwnerCanOnlyWithdraw() (gas: 19933)
[FAIL. Reason: EvmError: Revert] testWithdrawWithMultipleFunders() (gas: 19977)
```

I looked at the traces and it seems like all the tests that call the price feed contract from chainlink failed. I'm sure I got the price feed address right (https://sepolia.explorer.zksync.io/address/0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF). Could someone please assist me in figuring out why the tests are failing? Happy to provide more info if required. Thanks in advance!

EDIT: no errors on my alchemy!

**Comment by EngrPips** [comment]
Hello @spencerj411, working with zksync on `Foundry` is tricky and not straightforward because of how zksync is natively designed. For example, there are different types of system contracts that handle different operations on Zksync, such as nonce management, contract deployment, and so on. This is not the case on Ethereum. I will advise that you see the `Account Abstraction` course on Cyfrin Updraft first and then you can come back and try this.

**Reply by spencerj411** [reply]
Cheers for the response, @EngrPips. I don't think I quite get it based off your answer but I will take your advice, ignore it for now, and see later if I do!

**Reply by EngrPips** [reply]
When you reach the `Account Abstraction` part of the course, you will understand better what I mentioned here.

**Comment by spencerj411** [comment, marked as answer]
Update: I am currently watching the optional [Foundry Fundamentals Section 2: Foundry Fund Me 22. Zksync Devops](https://updraft.cyfrin.io/courses/foundry/foundry-fund-me/zksync-devops?lesson_format=video) lesson and from my understanding, some unit tests that usually passes when I test with an anvil or sepolia chain, fail due to compilation differences between zksync and evm.

**Reply by EngrPips** [reply]
Did those tests fail in the course, too? `ZkSync` is natively different from sepolia or Ethereum mainnet

**Reply by spencerj411** [reply]
yeah, some tests failed in the course as well.

**Reply by EngrPips** [reply]
Interesting
